sdk/src/Services/DynamoDBv2/Custom/DataModel/S3ClientCache.cs

```diff
@@ -34,12 +37,38 @@ internal void UseClient(ICoreAmazonS3 client, RegionEndpoint region)
             this.clientsByRegion.Add(region.SystemName, client);
         }
 
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026", 
+            Justification = "Reflection code is only used as a fallback in case the SDK was not trimmed. Trimmed scenarios should register dependencies with Amazon.RuntimeDependencyRegistry.GlobalRuntimeDependencyRegistry")]
         internal ICoreAmazonS3 GetClient(RegionEndpoint region)
         {
             ICoreAmazonS3 output;
+
             if (!this.clientsByRegion.TryGetValue(region.SystemName, out output))
             {
-                output = ServiceClientHelpers.CreateServiceFromAssembly<ICoreAmazonS3>(ServiceClientHelpers.S3_ASSEMBLY_NAME, ServiceClientHelpers.S3_SERVICE_CLASS_NAME, this.ddbClient);
+                output = GlobalRuntimeDependencyRegistry.Instance.GetInstance<ICoreAmazonS3>(ServiceClientHelpers.S3_ASSEMBLY_NAME, ServiceClientHelpers.S3_SERVICE_CLASS_NAME,
+                    new CreateInstanceContext(new S3ClientContext { Action = S3ClientContext.ActionContext.DynamoBDS3Link, Region = region }));
+
+                if (output == null)
+                {
+                    try
+                    {
+                        output = ServiceClientHelpers.CreateServiceFromAssembly<ICoreAmazonS3>(ServiceClientHelpers.S3_ASSEMBLY_NAME, ServiceClientHelpers.S3_SERVICE_CLASS_NAME, this.ddbClient);
+                    }
+                    catch (Exception e)
+                    {
+                        if (InternalSDKUtils.IsRunningNativeAot())
+                        {
+                            throw new MissingRuntimeDependencyException(ServiceClientHelpers.S3_ASSEMBLY_NAME, ServiceClientHelpers.S3_SERVICE_CLASS_NAME, nameof(GlobalRuntimeDependencyRegistry.RegisterS3Client));
+                        }
+
+                        var msg = string.Format(CultureInfo.CurrentCulture,
```

Comment:
Is this message correct? It says `STS_ASSEMBLY_NAME` and `AssumeRoleAWSCredentials`.

Reply:
Fixed, bad copy paste job.